velox/functions/prestosql/Slice.cpp

```diff
@@ -48,6 +48,9 @@ namespace {
 /// After adjustment, for the output ArrayVector:
 /// rawOffsets vector [1, 4, 8]
 /// rawSizes vector   [2, 2, 2]
+/// @tparam IndexKind The `TypeKind` of the function `Slice` start and length
```

Comment:
Maybe use `Kind` as the name because it controls both the type of index and length.